### PR TITLE
app-crypt/rnp new package rnp-0.15.0

### PR DESCRIPTION
**Pull request by antonsviridenko**
Signed-off-by: Anton Sviridenko <anton@picapica.im>

### CODE REVIEW

**Comment by scardracs** [comment]
Is this package required for package/s already on gentoo repo?

File: app-crypt/rnp/rnp-0.15.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+SRC_URI="https://codeload.github.com/rnpgp/rnp/tar.gz/refs/tags/v${PV} -> ${P}.tar.gz"
+
+LICENSE="BSD-2 BSD Apache-2.0"
+
+SLOT="0"
+
+KEYWORDS="~amd64 ~x86"
+
+IUSE="doc"
```
Comment (scardracs) [inline app-crypt/rnp/rnp-0.15.0.ebuild:20]: The newline should be only between SRC_URI and LICENSE

File: app-crypt/rnp/rnp-0.15.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+RDEPEND="
+		>=dev-libs/botan-2.14.0
+		>=dev-libs/json-c-0.11.0
+		app-arch/bzip2
+		sys-libs/zlib
```
Comment (scardracs) [inline app-crypt/rnp/rnp-0.15.0.ebuild:26]: Order alphabetically

**Comment by antonsviridenko** [comment]
@scardracs 

> Is this package required for package/s already on gentoo repo?

1. It can be used stand-alone, as a replacement for GnuPG. Package installs binaries (rnp and rnpkey).

2. RNP library is used by Mozilla Thunderbird since version 78, but seems that RNP it is bundled into the sources there now.

https://wiki.mozilla.org/Thunderbird:OpenPGP:Migration-From-Enigmail
https://hg.mozilla.org/comm-central/file/tip/third_party/README.rnp

**Comment by scardracs** [comment]
> @scardracs 
> 
> > Is this package required for package/s already on gentoo repo?
> 
> 1. It can be used stand-alone, as a replacement for GnuPG. Package installs binaries (rnp and rnpkey).
> 
> 2. RNP library is used by Mozilla Thunderbird since version 78, but seems that RNP it is bundled into the sources there now.
> 
> https://wiki.mozilla.org/Thunderbird:OpenPGP:Migration-From-Enigmail
> https://hg.mozilla.org/comm-central/file/tip/third_party/README.rnp

I'm asking because of section "New Packages" under Gentoo-bot

**Comment by antonsviridenko** [comment]
Yes it is a new package.

File: app-crypt/rnp/rnp-0.15.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	cmake_src_configure
+}
+
+src_compile() {
+	cmake_src_compile
+}
+
+src_install() {
+	cmake_src_install
+}
```
Comment (telans) [inline app-crypt/rnp/rnp-0.15.0.ebuild:48]: These are redundant with the cmake eclass already exporting these functions

```suggestion
```

File: app-crypt/rnp/rnp-0.15.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="high performance C++ OpenPGP library, fully compliant to RFC 4880"
```
Comment (telans) [inline app-crypt/rnp/rnp-0.15.0.ebuild:8]:
```suggestion
DESCRIPTION="High performance C++ OpenPGP library, fully compliant to RFC 4880"
```

File: app-crypt/rnp/rnp-0.15.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+HOMEPAGE="https://www.rnpgp.org/"
+SRC_URI="https://codeload.github.com/rnpgp/rnp/tar.gz/refs/tags/v${PV} -> ${P}.tar.gz"
```
Comment (telans) [inline app-crypt/rnp/rnp-0.15.0.ebuild:10]: No functional change but it matches just about every other ebuild

```suggestion
SRC_URI="https://github.com/rnpgp/rnp/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
```

File: app-crypt/rnp/rnp-0.15.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,48 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="high performance C++ OpenPGP library, fully compliant to RFC 4880"
+HOMEPAGE="https://www.rnpgp.org/"
```
Comment (telans) [inline app-crypt/rnp/rnp-0.15.0.ebuild:9]:
```suggestion
HOMEPAGE="https://www.rnpgp.org/ https://github.com/rnpgp/rnp/"
```

File: app-crypt/rnp/rnp-0.15.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+		sys-libs/zlib
+"
+
+DEPEND="${RDEPEND}"
+
+BDEPEND="
+		virtual/pkgconfig
+		>=dev-util/cmake-3.14.0
+		doc? ( dev-ruby/asciidoctor )
+"
```
Comment (telans) [inline app-crypt/rnp/rnp-0.15.0.ebuild:30]: Double tabs?

```suggestion
RDEPEND="
	app-arch/bzip2
	>=dev-libs/botan-2.14.0
	>=dev-libs/json-c-0.11.0
	sys-libs/zlib
"
DEPEND="${RDEPEND}"
BDEPEND="
	virtual/pkgconfig
	>=dev-util/cmake-3.14.0
	doc? ( dev-ruby/asciidoctor )
"
```

**Comment by antonsviridenko** [comment]
thank you @telans 

all suggested changes are pushed

**Comment by antonsviridenko** [comment]
Closing, new release ebuilds were put into the overlay instead
https://github.com/rnpgp/gentoo-rnp

**Pull request closed without merging**